### PR TITLE
Updating trigger handler to unset system fields if the record type is…

### LINKED ISSUE #1887: When Account Record Type is changed from Household Account, update _SYSTEM: fields
thread: https://powerofus.force.com/0D58000001YDFov

### LINKED ISSUE #2030: If Account Record Type is changed from the one set as the Household or One to One, update _SYSTEM fields so no longer acts as Household/One to One
Thread: https://powerofus.force.com/0D58000002bunro

### PR DESCRIPTION
**Pull request by njjc**
… not household or 1:1, and if the record is not the bucket account.
# Critical Changes

# Changes

# Issues Closed
- Fixes #1887 
- Fixes #2030

### CODE REVIEW

**Comment by njjc** [comment]
Not quite ready for review, this change causes a bunch of tests to fail for some reason.

File: src/classes/ACCT_IndividualAccounts_TEST.cls [hunk 1]
```diff
@@ -1738,4 +1738,45 @@ private class ACCT_IndividualAccounts_TEST {
...
+        Contact con = new Contact(
+            LastName = 'Test'
+        );
+        insert con;
+
+        Contact queryCon = [SELECT AccountId FROM Contact WHERE Id = :con.id];
+
+        Account queryAccount = [SELECT Id, RecordTypeId, npe01__SYSTEMISINDIVIDUAL__c, npe01__SYSTEM_AccountType__c FROM Account WHERE Id = :queryCon.AccountId];
+
+        System.assertEquals(rectypes[0].id, queryAccount.RecordTypeId, 'RecType should not be set to hh account rectype.');
```
Comment (davidhabib) [inline src/classes/ACCT_IndividualAccounts_TEST.cls]: error string is wrong.  it should be 'RecType should be set to....'

File: src/classes/ACCT_IndividualAccounts_TEST.cls [hunk 2]
```diff
@@ -1738,4 +1738,45 @@ private class ACCT_IndividualAccounts_TEST {
...
+        System.assertEquals(CAO_Constants.HH_ACCOUNT_TYPE, queryAccount.npe01__SYSTEM_AccountType__c, 'Account Type system field should be set for household account.');
+
+        Test.startTest();
+        queryAccount.RecordTypeId = rectypes[1].id;
+        update queryAccount;
+        Test.stopTest();
+
+        queryAccount = [SELECT Id, RecordTypeId, npe01__SYSTEMISINDIVIDUAL__c, npe01__SYSTEM_AccountType__c FROM Account WHERE Id = :queryCon.AccountId];
+
+        System.assertEquals(false, queryAccount.npe01__SYSTEMISINDIVIDUAL__c, 'Individual system field should be set for household account.');
```
Comment (davidhabib) [inline src/classes/ACCT_IndividualAccounts_TEST.cls]: error string should be 'Individual system field should false for non-household account'

File: src/classes/ACCT_IndividualAccounts_TEST.cls [hunk 3]
```diff
@@ -1738,4 +1738,45 @@ private class ACCT_IndividualAccounts_TEST {
...
+        System.assertNotEquals(CAO_Constants.HH_ACCOUNT_TYPE, queryAccount.npe01__SYSTEM_AccountType__c, 'Account Type system field should be set for household account.');
```
Comment (davidhabib) [inline src/classes/ACCT_IndividualAccounts_TEST.cls]: error string should be 'Account Type system field should not be set as household account'

**Comment by davidhabib** [comment]
@njjc the code and tests look fine, but the error strings on several of the asserts seems incorrect, and thus confusing as to what the intent is.  thanks.

File: src/classes/ACCT_IndividualAccounts_TEST.cls (hunk 1 above)
Comment (njjc) [inline src/classes/ACCT_IndividualAccounts_TEST.cls]: Thanks @davidhabib, fixed.

**Pull request merged**